idseq_dag/engine/pipeline_step.py

```diff
@@ -38,12 +38,24 @@ def __init__(self, name, input_files, output_files,
         self.upload_thread = None
         self.input_files_local = []
         self.additional_files_to_upload = []
+        self.counts_dict = {}
 
 
     @abstractmethod
     def run(self):
         ''' implement what is actually being run in this step '''
 
+    @abstractmethod
+    def count_reads(self):
+        ''' count reads '''
+
+    def save_counts(self):
+        if self.counts_dict:
+            count_file = "%s/%s.count" % (self.output_dir_local, self.name)
```

Comment:
nit readability suggestion: rename `count_file` to `count_filename`  and `countf` to `count_file`

Reply:
ok! thanks!